Update package properties to adhere to some best practices

cartermp opened the pull request:
Fixes https://github.com/fable-compiler/Fable/issues/2439 by adding:

* PackageLicenseExpression for the license
* Icon instead of IconURL
* SourceLink properties so that symbol packages (`.snupkg`) can also be published

Now when you pack each of the package directories, it will produce two packages under `bin/release`:

* `package-name.nupkg`
* `package-name.snupkg`

You'll want to upload both to NuGet when updating packages.

alfonsogarciacaro commented on src/Directory.Build.props:
```diff
@@ -2,9 +2,22 @@
+        <SymbolPackageFormat>snupkg</SymbolPackageFormat>
     </PropertyGroup>
+    <ItemGroup>
+        <None Include="..\..\fable_logo.png" Pack="true" PackagePath="\" />
```
It seems the fable_logo.png icon is missing from this repo, we should probably add it. It'd be also great if the directory is not relative (I guess we can use the MsBuildThisFileDirectory constant) so we can easily copy the file to other repos without depending on a specific structure.

cartermp replied:
Yep, it's included in the PR.

MangelMaxime replied:
Instead of using a relative link, I propose to use `$(MSBuildThisFileDirectory)`

If find it easier to read and it will works for all the packages in the repository independently of their level.

For example, in Fable.Form I do it like that for the license file

```xml
<PropertyGroup>
    <PackageLicenseFile>LICENSE.txt</PackageLicenseFile>
</PropertyGroup>
<ItemGroup>
    <None Include="$(MSBuildThisFileDirectory)\LICENSE.txt" Pack="true" PackagePath="$(PackageLicenseFile)" />
</ItemGroup>
```

cartermp replied:
Done.

alfonsogarciacaro reviewed (commented):
Thanks a lot @cartermp! If it's not too much to ask, would it be possible to also edit these lines in this PT so the publishing script automatically pushes the correct files as you mention in the description? https://github.com/fable-compiler/Fable/blob/caa715f1156be29c8dd9b866a03031a1852b3186/src/fable-publish-utils/PublishUtils.fs#L569-L588

cartermp commented:
@alfonsogarciacaro absolutely! Will take a look. It's starting to get into the evening now on my side, so I'll likely pick this back up tomorrow. I need to unwind a bit, otherwise I won't fall asleep easily!

nojaf commented on src/Directory.Build.props:
```diff
@@ -2,9 +2,22 @@
         <RepositoryUrl>https://github.com/fable-compiler/Fable.git</RepositoryUrl>
-        <PackageLicense>https://raw.githubusercontent.com/fable-compiler/Fable/master/LICENSE</PackageLicense>
-        <PackageIconUrl>https://raw.githubusercontent.com/fable-compiler/Fable/master/docs/img/fable_logo.png</PackageIconUrl>
+        <PackageLicenseExpression>MIT</PackageLicenseExpression>
```
@MangelMaxime would you prefer the actual license file here as well?

MangelMaxime replied:
It is true that I prefer the licence file in general.

My idea, is that using the licence file means that if a user read the code from Github or NuGet he will see the same source of truth ==> The same LICENSE.txt file

Related to one of this https://github.com/fable-compiler/fable-browser/pull/62 @nojaf sent

cartermp replied:
Done.

cartermp commented:
@alfonsogarciacaro @MangelMaxime Addressed all feedback. Since I don't have the ability to publish to NuGet, you'll have to verify the publish script works as expected, but I expect it will be fine.

Here's what the package will look like:

![image](https://user-images.githubusercontent.com/6309070/118019125-7becdf00-b30d-11eb-8b19-66b978b74193.png)

Note that determinism/compiler flags were always a red x mark, so nothing has regressed there. Critically, the icon, license, and symbols are all correct.

MangelMaxime commented:
Looks good to me 

@alfonsogarciacaro I let you publish the new versions 😉

alfonsogarciacaro commented on src/fable-publish-utils/PublishUtils.fs:
```diff
@@ -584,7 +584,21 @@ module Publish =
                 runList ["dotnet nuget push"; nupkg; "-s nuget.org -k"; nugetKey]
+
+                // Publish snupkg after the nupkg, since you need to already have a corresponding nupkg to match the symbols against
+                runList ["dotnet nuget push"; snupkg; "-s nuget.org -k"; nugetKey]
```
@cartermp I got this when trying to push a package so it seems the first `dotnet nuget push` command is already detecting and pushing the symbols (which makes the second command fail because they're already published) so I have commented out this line.

```
> dotnet nuget push src/Fable.AST/temp/Fable.AST.3.1.1.nupkg -s nuget.org -k ...
Pushing Fable.AST.3.1.1.nupkg to 'https://www.nuget.org/api/v2/package'...
  PUT https://www.nuget.org/api/v2/package/
  Created https://www.nuget.org/api/v2/package/ 1690ms
Your package was pushed.
Pushing Fable.AST.3.1.1.snupkg to 'https://www.nuget.org/api/v2/symbolpackage'...
  PUT https://www.nuget.org/api/v2/symbolpackage/
  Created https://www.nuget.org/api/v2/symbolpackage/ 725ms
Your package was pushed.
> dotnet nuget push src/Fable.AST/temp/Fable.AST.3.1.1.snupkg -s nuget.org -k ...
Pushing Fable.AST.3.1.1.snupkg to 'https://www.nuget.org/api/v2/symbolpackage'...
  PUT https://www.nuget.org/api/v2/symbolpackage/
  Conflict https://www.nuget.org/api/v2/symbolpackage/ 843ms
To skip already published packages, use the option --skip-duplicate
```

cartermp replied:
Makes sense! Interesting that it all just works with the snupkg.

alfonsogarciacaro replied:
If we're not sure what are the conditions to trigger the symbol publishing maybe it's better to still issue the second command with `-skip-duplicate` just to be sure 🤔
